libraries/common/io/files.effekt

```diff
@@ -55,51 +29,27 @@ type Mode {
   ReadWriteExclusive()  // 'wx+' created if does not exist, truncated if exists, fails if exists
 }
 
-def modeName(modes: Mode): String = modes match {
-  case Append()              => "a"
-  case ReadAppend()          => "a+"
-  case ReadOnly()            => "r"
-  case ReadWrite()           => "w+"
-  case Write()               => "w"
-  case AppendExclusive()     => "ax"
-  case ReadAppendExclusive() => "ax+"
-  case AppendSync()          => "as"
-  case ReadAppendSync()      => "as+"
-  case ReadSync()            => "rs"
-  case ReadWriteSync()       => "rs+"
-  case WriteExclusive()      => "wx"
-  case ReadWriteExclusive()  => "wx+"
-}
-
 
 /**
- * Using the file system via node.js
+ * A file descriptor. Should not be inspected.
  */
-def filesystem[R] { program: => R / Files }: R / { AsyncIO, Exception[IOError] } = // TODO move Exception to be bidirectional
-  try { program() }
-  with Files {
-    def readFile(path) = resume(readFile(path))
-    def writeFile(path, contents) = resume(writeFile(path, contents))
-  }
+type File = Int
 
 
-// AsyncIO-based API (default)
-// ---------------------------
-
 /**
  * Reads a file at given path as utf8 encoded string.
  */
-def readFile(path: String): String / { AsyncIO, Exception[IOError] } = {
-  val fd = openFile(path, ReadOnly());
-  with on[IOError].finalize { closeFile(fd) }
+def readFile(path: String): String / Exception[IOError] = {
+  val fd = direct::open(path, ReadOnly());
+  with on[IOError].finalize { direct::close(fd); () }
```

Comment:
Note, that if you don't hide it behind the namespace `direct`, you can call it `fd.close()`.